From fdbabd22501e1fffe0267e8077c2d99b941edfd3 Mon Sep 17 00:00:00 2001
From: Suraj P Kizhakkethil <quic_surapk@quicinc.com>
Date: Mon, 23 Dec 2024 13:49:06 +0530
Subject: [PATCH] QSDK:CP:mac80211.sh: Remove deprecated code

Remove duplicate mbssid config in hostapd config file.
Primary link feature is added in pre_post_wifi_operations.sh. Remove deprecated/obsolete code in mac80211.sh

Change-Id: If818316f03a3da76a33d82606229085eafb0855f
Signed-off-by: Suraj P Kizhakkethil <quic_surapk@quicinc.com>
---
 .../files/lib/netifd/wireless/mac80211.sh     | 24 -------------------
 1 file changed, 24 deletions(-)

diff --git a/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh b/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
index c00c27d636a0..fd134bb68a0a 100755
--- a/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
+++ b/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
@@ -801,7 +801,6 @@ ${channel:+channel=$channel}
 ${channel_list:+chanlist=$channel_list}
 ${hostapd_noscan:+noscan=1}
 ${tx_burst:+tx_queue_data2_burst=$tx_burst}
-${multiple_bssid:+mbssid=$multiple_bssid}
 #num_global_macaddr=$num_global_macaddr
 #macaddr_base=$macaddr_base
 $base_cfg
@@ -1533,7 +1532,6 @@ wpa_supplicant_set_config() {
 	[ "$ret" != 0 -o -z "$supplicant_res" ] && wireless_setup_vif_failed WPA_SUPPLICANT_FAILED
 
 	wireless_add_process "$(jsonfilter -s "$supplicant_res" -l 1 -e @.pid)" "/usr/sbin/wpa_supplicant" 1 1
-	update_primary_link
 }
 
 hostapd_set_config() {
@@ -1915,28 +1913,6 @@ drv_mac80211_teardown() {
 	mac80211_reset_config "$phy"
 }
 
-update_primary_link()
-{
-        if [ -n "$mld_names" ]; then
-                for mld_iface in $mld_names; do
-                        config_get mld_primary_link "$mld_iface" primary_link
-                        config_get mld_ifname "$mld_iface" ifname
-                        if [ -n "$mld_primary_link" ]; then
-                                while true;
-                                do
-                                        ifname_state="$(hostapd_cli -i "$mld_ifname" status 2> /dev/null | grep state | cut -d'=' -f 2)"
-                                        if [ "$ifname_state" = "ENABLED" ]; then
-                                                echo "$mld_primary_link" > /sys/kernel/debug/ieee80211/phy"${phy#phy}"/netdev:"$mld_ifname"/primary_link
-                                                break;
-                                        fi
-                                done
-                        fi
-                done
-        fi
-
-}
-
-
 _sta_radios=
 mac80211_derive_ml_info() {
 	local _mlds
-- 
2.34.1

